File: Creational/Builder/Java/AbstractProduct.java
import java.util.HashSet;
import java.util.Set;

public abstract class AbstractProduct {
	
	protected Set<String> string_set;
	
	protected AbstractProduct() {
		this.string_set = new HashSet<>();
	}
	
	public void operation() {
		
		for (String string : string_set) {
			System.out.print(string + " ");
		}
		
		System.out.println();
	}
	
	public abstract void stepA1();
	public abstract void stepA2();
	public abstract void stepB1();
	public abstract void stepB2();
	public abstract void stepC1();
	public abstract void stepC2();

}

File: Creational/Builder/Kotlin/AbstractProduct.kt
abstract class AbstractProduct {

    protected var stringSet = mutableSetOf<String>()

    fun operation() {
        println(this.stringSet.toString())
    }

    abstract fun stepA1()
    abstract fun stepA2()
    abstract fun stepB1()
    abstract fun stepB2()
    abstract fun stepC1()
    abstract fun stepC2()

}